app/views/webhooks/index.html.erb

```diff
@@ -3,48 +3,129 @@
 <%= render 'projects/header', project: @project, tab: "webhooks" %>
 
 <% inbound_webhooks = @project.webhooks %>
-<% outbound_webhooks = @project.outbound_webhooks %>
-
 <section class="tabs">
-  <h2>Inbound Webhooks</h2>
-  <p>In order to automatically trigger a deploy to a stage, add a branch &rarr; stage mapping below
-  and paste in the project's webhook URL to your CI service of choice.</p>
+  <h2>
+    Inbound Webhooks
+    <%= additional_info "Allows CI or source-code provider to trigger a release or deploy." %>
+  </h2>
+
+  <details>
+    <summary>Callback URLs</summary>
 
-  <dl>
-    <% Samson::Integration::SOURCES.each do |source| %>
-      <dt>
-        <%= source.titleize %><%= webhook_help_text source %>
-      </dt>
-      <dd><%= send("integrations_#{source}_deploy_url", @project.token) %>
-        <% if source == 'github' && token = Integrations::GithubController.secret_token %>
-          <br/>
-          <% if current_user.admin_for?(@project) %>
-            (Set <%= link_to "secret token", "javascript:prompt('Copy this token', '#{token}')" %> in <%= link_to 'hook settings', "https://github.com/#{@project.repository_path}/settings/hooks" %>)
+    <dl>
+      <% Samson::Integration::SOURCES.each do |source| %>
+        <dt>
+          <% if source == 'generic' %>
+            <details>
+              <summary><%= source.titleize %></summary>
+              Generic endpoint to start deploys, expects post payload in the form of:
+              <pre><%= <<~TEXT
+                {
+                  deploy: {
+                    branch: &ltname&gt,
+                    commit: {
+                      sha: &ltsha&gt,
+                      message: &ltmessage&gt
+                    }
+                  }
+                }
+                curl URL -H 'Content-Type: application/json' -d '{"deploy":{"branch":"master","commit":{"sha":"abc","message":"hi"}}}'
```

Comment:
separate code block for json and curl?

Reply:
either that or use 
```
cat << EOF | curl -d - ...
{
  "deploy": {}
}
EOF
```

Reply:
also interpolate the URL.. we should know it right?

Reply:
change to just curl + real url